specutils/spectra/spectrum1d.py

```diff
@@ -60,6 +60,12 @@ def __init__(self, flux=None, spectral_axis=None, wcs=None,
         if flux is None and 'data' in kwargs:
             flux = kwargs.pop('data')
 
+        # Check to make sure the wavelength length is the same in both
```

Comment:
This block should occur after the parsing of the spectral axis and flux values (the if/else block starting at line 77). This is because of the way the arithmetic works on nddata objects -- it's possible that `flux` is a single float value since operands get converted to the nddata subclass (i.e. `my_spec * 15`, `15` is converted to an instance of `Spectrum1D`).

Reply:
I would suggest to complement this with a check for the length/shape of the `uncertainty` property, if set, e.g.
```
        if flux is not None and self.uncertainty is not None:
            if self.uncertainty.array.shape != flux.shape:
                raise ValueError('Flux ({}) and uncertainty ({}) shapes must match'.format(
                    flux.shape, self.uncertainty.array.shape))
```
(probably needs to be inserted after the final `super(Spectrum1D, self).__init__`). Don't know if the above catches all valid cases, or if, say, a single value for `uncertainty` (shape `()`) should also be accepted.

Reply:
hopefully the order is better now.
added in ck for uncertainty shape too